httpx/dispatch/connection.py

```diff
@@ -70,10 +72,10 @@ def __init__(
         self,
         verify: VerifyTypes = None,
         cert: CertTypes = None,
-        timeout: TimeoutTypes = None,
+        timeout: TimeoutTypes = UNSET,
     ) -> None:
         ssl = self.ssl.with_overrides(verify=verify, cert=cert)
-        timeout = self.timeout if timeout is None else TimeoutConfig(timeout)
+        timeout = TimeoutConfig(timeout if timeout is not UNSET else self.timeout)
```

Comment:
Shouldn't we keep the same logic as before?

```suggestion
        timeout = self.timeout if timeout is UNSET else TimeoutConfig(timeout)
```

Reply:
Reverted